### ISSUE TITLE
Carousel: a11y fix - clipped heading

### ISSUE THREAD

**Issue by sendlo** [eBay/ebayui-core#325]
# Bug Report

## Description
Add Support for Clipped Heading
Final issue from: https://github.com/eBay/ebayui-core/issues/294

We should also allow the option of this outer span being a heading element too (h1 - h6), in the case where the page/app does not have a visible heading preceding the carousel.

```html
<span id="carousel-status-c1-1" class="clipped" role="status" aria-live="polite">
  <span>Showing slide 1 of 4 - Top Products - Carousel</span>
</span>
```

Could be:

```html
<h2 id="carousel-status-c1-1" class="clipped" role="status"  aria-live="polite">
  <span>Showing slide 1 of 4 - Top Products - Carousel</span>
</h2>
```

Reason: all carousels should be preceded by a heading.

**Comment by DylanPiercey** [comment]
Since the status is clipped/hidden do we still need to provide the ability to change the span to any heading? Or would just using an h2 suffice? This way we could have a boolean attribute like `a11y-heading` or `no-heading` instead of just a pass through tag name.
